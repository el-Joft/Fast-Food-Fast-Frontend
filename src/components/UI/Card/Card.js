/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getAllMenus } from '../../../actions/getMenusAction';
import { addCart } from '../../../actions/orderAction';


class Card extends Component {
  componentDidMount() {
    this.props.getAllMenus();
  }

  addToCart(menuData) {
    const item = Object.assign({}, menuData, { quantity: 1 });
    this.props.addCart(item);
  }

  renderCard = () => {
    const { menu } = this.props;
    if (menu.menu !== null) {
      return menu.menu.map(item => (
        <div key={item.id} className='menu-section'>
          <div className="menu-food">
            <h3>{item.name}</h3>
            <p>{item.description}</p>
          </div>
          <div className="menu-price">
            <p>
              <strike>N</strike>
              {item.price}
            </p>
            <div to="/checkout">
              {' '}
              <span
                role="searchbox"
                onKeyPress={() => this.addToCart(item)}
                onClick={menuData => this.addToCart(item)}
                tabIndex="-1"
                className="menu-order"
              >
                Make Order
              </span>
            </div>
          </div>

        </div>
      ));
    }
  }

  render() {
    return (
      <div className="menu-body">
        {this.renderCard()}
      </div>
    );
  }
}

Card.propTypes = {
  getAllMenus: PropTypes.func.isRequired,
  addCart: PropTypes.func.isRequired,
  menu: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
};

// we get values here using the this.props.errors/auth

const mapStateToProps = state => ({
  menu: state.menu,
  errors: state.errors,
  order: state.order,
});

export default connect(mapStateToProps, { getAllMenus, addCart })(Card);
