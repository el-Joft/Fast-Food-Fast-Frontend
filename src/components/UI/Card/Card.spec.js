/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { Card } from './Card';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      getAllMenus: jest.fn(),
      addCart: jest.fn(),
      history: '',
      success: {
        status: true,
      },
      menu: {
        menu: [],
      },
      auth: {
        isAuthenticated: true,
        user: {
          id: 1,
        },
      },
    };
  });

  it('invokes `componentDidMount` when mounted', () => {
    const spy = jest.spyOn(Card.prototype, 'componentDidMount');
    wrapper = shallow(<Card {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should render the component', () => {
    wrapper = shallow(<Card {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Button component', () => {
    wrapper = shallow(<Card {...props} />);
    expect(wrapper.length).toBe(1);
  });
});
