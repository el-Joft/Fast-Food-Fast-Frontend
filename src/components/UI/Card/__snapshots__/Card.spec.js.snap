// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addCart={[MockFunction]}
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "id": 1,
        },
      }
    }
    getAllMenus={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    history=""
    menu={
      Object {
        "menu": Array [],
      }
    }
    success={
      Object {
        "status": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "menu-body",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "menu-body",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
